src/AppInstallerCLICore/Commands/COMInstallCommand.cpp

```diff
@@ -14,114 +12,11 @@ using namespace AppInstaller::Utility::literals;
-        };
-    }
-
-    Resource::LocString COMInstallCommand::ShortDescription() const
-    {
-        return { Resource::String::InstallCommandShortDescription };
-    }
-
-    Resource::LocString COMInstallCommand::LongDescription() const
-    {
-        return { Resource::String::InstallCommandLongDescription };
-    }
-
-    void COMInstallCommand::Complete(Context& context, Args::Type valueType) const
-    {
-        switch (valueType)
-        {
-        case Args::Type::Query:
-        case Args::Type::Manifest:
-        case Args::Type::Id:
-        case Args::Type::Name:
-        case Args::Type::Moniker:
-        case Args::Type::Version:
-        case Args::Type::Channel:
-        case Args::Type::Source:
-            context <<
-                Workflow::CompleteWithSingleSemanticsForValue(valueType);
-            break;
-        case Args::Type::Locale:
-            // May well move to CompleteWithSingleSemanticsForValue,
-            // but for now output nothing.
-            context <<
-                Workflow::CompleteWithEmptySet;
-            break;
-        case Args::Type::Log:
-        case Args::Type::InstallLocation:
-            // Intentionally output nothing to allow pass through to filesystem.
-            break;
-        }
-    }
-
-    std::string COMInstallCommand::HelpLink() const
-    {
-        return "https://aka.ms/winget-command-install";
-    }
-
-    void COMInstallCommand::ValidateArgumentsInternal(Args& execArgs) const
-    {
-        if (execArgs.Contains(Args::Type::Manifest) &&
-            (execArgs.Contains(Args::Type::Query) ||
-                execArgs.Contains(Args::Type::Id) ||
-                execArgs.Contains(Args::Type::Name) ||
-                execArgs.Contains(Args::Type::Moniker) ||
-                execArgs.Contains(Args::Type::Version) ||
-                execArgs.Contains(Args::Type::Channel) ||
-                execArgs.Contains(Args::Type::Source) ||
-                execArgs.Contains(Args::Type::Exact)))
-        {
-            throw CommandException(Resource::String::BothManifestAndSearchQueryProvided);
-        }
-
-        if (execArgs.Contains(Args::Type::InstallScope))
-        {
-            if (ConvertToScopeEnum(execArgs.GetArg(Args::Type::InstallScope)) == Manifest::ScopeEnum::Unknown)
-            {
-                throw CommandException(Resource::String::InvalidArgumentValueError, s_ArgumentName_Scope, { "user"_lis, "machine"_lis });
-            }
-        }
-
-        if (execArgs.Contains(Args::Type::Locale))
-        {
-            if (!Locale::IsWellFormedBcp47Tag(execArgs.GetArg(Args::Type::Locale)))
-            {
-                throw CommandException(Resource::String::InvalidArgumentValueErrorWithoutValidValues, Argument::ForType(Args::Type::Locale).Name(), {});
-            }
-        }
-    }
-
+    // IMPORTANT: To use this command, the caller should have already retrieved the package manifest (GetManifest()) and added it to the Context Data
```

Comment:
I really meant a very precise set of context data items; this does not specify that.  It simply says "put a manifest in" which could theoretically be into any data item that can contain a manifest.  There could be more than one in the future.